lambdas/src/main/kotlin/scorcerer/server/resources/MatchResource.kt

```diff
@@ -20,4 +27,22 @@ class MatchResource : MatchApi() {
     override fun setMatchScore(requesterUserId: String, matchId: String, setMatchScoreRequest: SetMatchScoreRequest) {
         TODO("Not yet implemented")
     }
+
+    override fun createMatch(requesterUserId: String, createMatchRequest: CreateMatchRequest): CreateMatch200Response {
+        val id = try {
+            transaction {
+                MatchTable.insert {
+                    it[this.homeTeamId] = createMatchRequest.homeTeamId
+                    it[this.awayTeamId] = createMatchRequest.awayTeamId
+                    it[this.datetime] = createMatchRequest.datetime
+                    it[this.state] = MatchState.UPCOMING
+                    it[this.venue] = createMatchRequest.venue
+                    it[this.matchDay] = createMatchRequest.matchDay
+                } get MatchTable.id
+            }
+        } catch (e: PSQLException) {
+            throw e
```

Comment:
There’s not much value in catching the error and then just throwing it again. We can either remove the `try` or throw an `ApiErrorResponse` here

Reply:
Ahh yeah good point